internal/config/config.go

```diff
@@ -173,14 +178,18 @@ func (c *Config) ToFakeGcsOptions() fakestorage.Options {
 			}
 		}
 	}
+	port := c.Port
+	if scheme == "http" {
+		port = c.PortHTTP
+	}
```

Comment:
Won't this be a breaking change? Like, if I'm currently using fake-gcs-server like this:

```
fake-gcs-server -scheme http -port 8080
```

The server will now start on port `8000` instead of `8080`.

I know this is fairly complicated, but could we make it so:

1. `port` and `port-http` no longer have a default at the flag definition. We'll handle it ourselves
2. if the user specifies `-scheme http` and `-port`, we use that port (keep current behavior)
3. if the user specifies just `-scheme http`, we use the default `-port` (keep current behavior)
4. if the user specifies just `-scheme both`, we'll use the default `-port` and `-port-http`
5. if the user doesn't provide any flags, we'll use the default `-port` and `-scheme https`

Perhaps we can turn each of the 5 examples in test cases for the function/core logic that handles this.

We may be able to implement this custom logic with a custom implementation of [flag.Value](https://pkg.go.dev/flag#Value), but it sounds tricky as it'd need to be aware of the value of another flag. The biggest advantage of using a custom flag.Value is differentiating between the user not providing the flag and setting it to `0` (I'm not super concerned about that though - right now if the user sets it to `0`, we start the server in a random port, but never print that port to the user, so who cares? heh)

Reply:
You are correct. I overlooked the http port in my last fix and broke the behavior, that I previously maintained. Thanks for catching that!

I implemented the `port` and `port-http` without defaults in the `flag` API but set it myself in the code if their value is 0. As you suggest, I also think that no one uses it today with port 0, so I don't think it is a breaking change, so I didn't implement the custom `flag.Value`.
I have an idea of how to implement it, maybe using `flag.Visit`, and I will do it if you think it is needed.
Please let me know. 

Will work on adding the test cases soon.

Reply:
@fsouza 
Added unit tests to cover the examples you listed and few more to be on the safe side. The tests found an issue with `both` scheme in the `cfg.externalURL` config - fixed.
Added 2 more runs to the github action to cover `both` scheme with default and non-default ports.

Let me know what you think. Thanks